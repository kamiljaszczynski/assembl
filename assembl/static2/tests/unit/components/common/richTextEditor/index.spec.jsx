import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { ContentState, EditorState, RichUtils } from 'draft-js';

import RichTextEditor from '../../../../../js/app/components/common/richTextEditor';

describe('RichTextEditor component', () => {
  it.skip('should render a rich text editor', () => {
    const editorState = EditorState.createEmpty();
    const updateEditorStateSpy = jest.fn((newState) => {
      return newState;
    });
    const props = {
      editorState: editorState,
      maxLength: 2000,
      placeholder: 'Write here',
      updateEditorState: updateEditorStateSpy
    };

    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<RichTextEditor {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  describe('getCharCount method', () => {
    it('should return 0 if the editorState content is empty', () => {
      const rte = new RichTextEditor();
      const actual = rte.getCharCount(EditorState.createEmpty());
      expect(actual).toEqual(0);
    });

    it('should return the number of characters in the editorState', () => {
      const editorState = EditorState.createWithContent(ContentState.createFromText('Hello world'));
      const rte = new RichTextEditor();
      const actual = rte.getCharCount(editorState);
      expect(actual).toEqual(11);
    });
  });

  describe('shouldHidePlaceholder method', () => {
    it('should return true if the editorState has no content and the block type is not unstyled', () => {
      const editorState = EditorState.createEmpty();
      const newEditorState = RichUtils.toggleBlockType(editorState, 'unordered-list-item');
      const rte = new RichTextEditor();
      rte.props = {
        editorState: newEditorState
      };
      const actual = rte.shouldHidePlaceholder();
      expect(actual).toBeTruthy();
    });

    it('should return false if the editorState has content', () => {
      const editorState = EditorState.createWithContent(ContentState.createFromText('Hello world'));
      const rte = new RichTextEditor();
      rte.props = {
        editorState: editorState
      };
      const actual = rte.shouldHidePlaceholder();
      expect(actual).toBeFalsy();
    });

    it('should return false if the block type is unstyled', () => {
      const editorState = EditorState.createEmpty();
      const rte = new RichTextEditor();
      rte.props = {
        editorState: editorState
      };
      const actual = rte.shouldHidePlaceholder();
      expect(actual).toBeFalsy();
    });
  });
});