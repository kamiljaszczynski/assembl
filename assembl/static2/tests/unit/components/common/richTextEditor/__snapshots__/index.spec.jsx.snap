// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichTextEditor component should render a rich text editor 1`] = `
<div
  className="rich-text-editor"
>
  <Toolbar
    buttonsConfig={
      Array [
        Object {
          "icon": "text-bold",
          "id": "bold",
          "label": "Bold",
          "style": "BOLD",
          "type": "style",
        },
        Object {
          "icon": "text-italics",
          "id": "italic",
          "label": "Italic",
          "style": "ITALIC",
          "type": "style",
        },
        Object {
          "icon": "text-bullets",
          "id": "bullets",
          "label": "Bullet List",
          "style": "unordered-list-item",
          "type": "block-type",
        },
      ]
    }
    editorState={
      EditorState {
        "_immutable": Object {
          "allowUndo": true,
          "currentContent": Object {
            "blockMap": Object {
              "9iel9": Object {
                "characterList": Array [],
                "data": Object {},
                "depth": 0,
                "key": "9iel9",
                "text": "",
                "type": "unstyled",
              },
            },
            "entityMap": Object {
              "__add": [Function],
              "__create": [Function],
              "__get": [Function],
              "__getLastCreatedEntityKey": [Function],
              "__mergeData": [Function],
              "__replaceData": [Function],
              "add": [Function],
              "create": [Function],
              "get": [Function],
              "getLastCreatedEntityKey": [Function],
              "mergeData": [Function],
              "replaceData": [Function],
            },
            "selectionAfter": Object {
              "anchorKey": "9iel9",
              "anchorOffset": 0,
              "focusKey": "9iel9",
              "focusOffset": 0,
              "hasFocus": false,
              "isBackward": false,
            },
            "selectionBefore": Object {
              "anchorKey": "9iel9",
              "anchorOffset": 0,
              "focusKey": "9iel9",
              "focusOffset": 0,
              "hasFocus": false,
              "isBackward": false,
            },
          },
          "decorator": null,
          "directionMap": Object {
            "9iel9": "LTR",
          },
          "forceSelection": false,
          "inCompositionMode": false,
          "inlineStyleOverride": null,
          "lastChangeType": null,
          "nativelyRenderedContent": null,
          "redoStack": Array [],
          "selection": Object {
            "anchorKey": "9iel9",
            "anchorOffset": 0,
            "focusKey": "9iel9",
            "focusOffset": 0,
            "hasFocus": false,
            "isBackward": false,
          },
          "treeMap": Object {
            "9iel9": Array [
              Object {
                "decoratorKey": null,
                "end": 0,
                "leaves": Array [
                  Object {
                    "end": 0,
                    "start": 0,
                  },
                ],
                "start": 0,
              },
            ],
          },
          "undoStack": Array [],
        },
      }
    }
    focusEditor={[Function]}
    onChange={[Function]}
  />
  <DraftEditor
    blockRenderMap={
      Immutable.Map {
        ordered-list-item: Object {
          "element": "li",
          "wrapper": <ol
            className="public-DraftStyleDefault-ol"
        />,
        },
        header-six: Object {
          "element": "h6",
        },
        header-four: Object {
          "element": "h4",
        },
        header-one: Object {
          "element": "h1",
        },
        unordered-list-item: Object {
          "element": "li",
          "wrapper": <ul
            className="public-DraftStyleDefault-ul"
        />,
        },
        atomic: Object {
          "element": "figure",
        },
        unstyled: Object {
          "aliasedElements": Array [
            "p",
          ],
          "element": "div",
        },
        header-two: Object {
          "element": "h2",
        },
        code-block: Object {
          "element": "pre",
          "wrapper": <pre
            className="public-DraftStyleDefault-pre"
        />,
        },
        blockquote: Object {
          "element": "blockquote",
        },
        header-five: Object {
          "element": "h5",
        },
        header-three: Object {
          "element": "h3",
        },
      }
    }
    blockRendererFn={[Function]}
    blockStyleFn={[Function]}
    editorState={
      EditorState {
        "_immutable": Object {
          "allowUndo": true,
          "currentContent": Object {
            "blockMap": Object {
              "9iel9": Object {
                "characterList": Array [],
                "data": Object {},
                "depth": 0,
                "key": "9iel9",
                "text": "",
                "type": "unstyled",
              },
            },
            "entityMap": Object {
              "__add": [Function],
              "__create": [Function],
              "__get": [Function],
              "__getLastCreatedEntityKey": [Function],
              "__mergeData": [Function],
              "__replaceData": [Function],
              "add": [Function],
              "create": [Function],
              "get": [Function],
              "getLastCreatedEntityKey": [Function],
              "mergeData": [Function],
              "replaceData": [Function],
            },
            "selectionAfter": Object {
              "anchorKey": "9iel9",
              "anchorOffset": 0,
              "focusKey": "9iel9",
              "focusOffset": 0,
              "hasFocus": false,
              "isBackward": false,
            },
            "selectionBefore": Object {
              "anchorKey": "9iel9",
              "anchorOffset": 0,
              "focusKey": "9iel9",
              "focusOffset": 0,
              "hasFocus": false,
              "isBackward": false,
            },
          },
          "decorator": null,
          "directionMap": Object {
            "9iel9": "LTR",
          },
          "forceSelection": false,
          "inCompositionMode": false,
          "inlineStyleOverride": null,
          "lastChangeType": null,
          "nativelyRenderedContent": null,
          "redoStack": Array [],
          "selection": Object {
            "anchorKey": "9iel9",
            "anchorOffset": 0,
            "focusKey": "9iel9",
            "focusOffset": 0,
            "hasFocus": false,
            "isBackward": false,
          },
          "treeMap": Object {
            "9iel9": Array [
              Object {
                "decoratorKey": null,
                "end": 0,
                "leaves": Array [
                  Object {
                    "end": 0,
                    "start": 0,
                  },
                ],
                "start": 0,
              },
            ],
          },
          "undoStack": Array [],
        },
      }
    }
    keyBindingFn={[Function]}
    onChange={[Function]}
    placeholder="Write here"
    readOnly={false}
    spellCheck={false}
    stripPastedStyles={false}
  />
  <div
    className="annotation margin-xs"
  >
    <Translate
      nbCharacters={2000}
      value="debate.remaining_x_characters"
    />
  </div>
</div>
`;
